ckanext/canada/templates/snippets/cdts/header.html

```diff
@@ -0,0 +1,259 @@
+{% set lang = h.lang() %}
+
+{% macro recombinant_menu_text(r_type) %}
+  {{- _(h.recombinant_get_geno(r_type).shortname or h.recombinant_get_geno(r_type).title) -}}
+{% endmacro %}
+
+{% macro recombinant_menu_link(r_type) %}
+  {{- h.url_for('recombinant.type_redirect', resource_name=r_type) -}}
+{% endmacro %}
+
+{# See: https://github.com/wet-boew/cdts-sgdc/blob/v4.1.0/src/gcweb/appTop.ejs #}
+{# See: https://github.com/wet-boew/cdts-sgdc/blob/v4.1.0/src/gcweb/top.ejs #}
+<script>
+  let appTop = document.getElementById("app-top");
+  if( appTop ){
+    {% if g.is_registry %}
+      const menuLinks = [
+        {% if g.userobj %}
+          {
+            "href": "{{ h.url_for('canada.links') }}",
+            "text": "{{ _('Home') }}",
+          },
+        {% else %}
+          {
+            "href": "{{ h.url_for('user.login') }}",
+            "text": "{{ _('Home') }}",
+          },
+        {% endif %}
+        {% if h.check_access('package_create') %}
+          {
+            "text": "{{ _('Open Data') }}",
+            "id": "open-data",
+            "subLinks": [
+              {
+                "subhref": "{{ h.url_for('dataset_new') }}",
+                "subtext": "{{ _('Create an Open Data Record') }}",
```

Comment:
all of these variables that are inserting values into json will be escaped by jinja2 like they are HTML. Instead of `"{{...}}"` we should use `{{ ...|tojson }}` for correct escaping.

```suggestion
                "subhref": {{ h.url_for('dataset_new')|tojson }},
                "subtext": {{ _('Create an Open Data Record')|tojson }},
```

Reply:
Cool. @wardi this is done now for all the calls to gettext and helper methods etc in the CDTS headers.html here.